### PR TITLE
Fulfillment confirmation

### PR DESCRIPTION
**Pull request by tomaszszymanski129**
I want to merge this change because it merges feature `Fulfillment confirmation` to master.
Each of PR this one is created from have already been reviewed separately.

PRs which this branch is build from:
https://github.com/mirumee/saleor/pull/7645 https://github.com/mirumee/saleor/pull/7646 https://github.com/mirumee/saleor/pull/7665 https://github.com/mirumee/saleor/pull/7679 https://github.com/mirumee/saleor/pull/7725 https://github.com/mirumee/saleor/pull/7726

⚠️ Should be merged together with frontend.

# Impact

* [x] New migrations
* [x] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [x] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [x] Database migration files are up to date
* [ ] The changes are tested
* [x] GraphQL schema and type definitions are up to date
* [x] Changes are mentioned in the changelog

### CODE REVIEW

File: saleor/graphql/order/mutations/fulfillments.py [hunk 1]
```diff
@@ -316,6 +331,56 @@ def perform_mutation(cls, _root, info, **data):
...
+    class Meta:
+        description = "Approve existing fulfillment."
+        permissions = (OrderPermissions.MANAGE_ORDERS,)
+        error_type_class = OrderError
+        error_type_field = "order_errors"
+
+    @classmethod
+    def clean_input(cls, info, fulfillment):
+        if fulfillment.status != FulfillmentStatus.WAITING_FOR_APPROVAL:
+            raise ValidationError(
```
Comment (korycins) [inline saleor/graphql/order/mutations/fulfillments.py:353]: `ValidationError("field":ValdiationError()}`?

File: saleor/order/actions.py [hunk 2]
```diff
@@ -319,6 +337,42 @@ def cancel_fulfillment(
...
+            line=f_line.order_line,
+            quantity=f_line.quantity,
+            variant=f_line.order_line.variant,
+            warehouse_pk=str(f_line.stock.warehouse_id),  # type: ignore
+        )
+        for f_line in fulfillment.lines.all()
+    ]
+    fulfill_order_lines(lines_to_fulfill)
+    order.refresh_from_db()
+    manager.order_updated(order)
```
Comment (korycins) [inline saleor/order/actions.py:369]: All manager actions here  should be called in `transaction.on_commit`

File: saleor/order/actions.py [hunk 3]
```diff
@@ -237,6 +237,24 @@ def order_fulfilled(
...
+    fulfillment_lines: List["FulfillmentLine"],
+    manager: "PluginsManager",
+    _notify_customer=True,
+):
+    order = fulfillments[0].order
+    update_order_status(order)
+    events.fulfillment_awaits_approval_event(
+        order=order, user=user, app=app, fulfillment_lines=fulfillment_lines
+    )
+    manager.order_updated(order)
```
Comment (korycins) [inline saleor/order/actions.py:254]: Duplicated

**Review by karola312 — APPROVED** [review]

**Comment by Khnsimeon29** [comment]
Hello

**Review by maarcingebala — APPROVED** [review]

**Pull request merged**